src/TLuaInterpreter.cpp

```diff
@@ -2801,15 +2801,17 @@ int TLuaInterpreter::setFont(lua_State* L)
 
     if (windowName.isEmpty() || windowName.compare(QStringLiteral("main"), Qt::CaseSensitive) == 0) {
         if (mudlet::self()->mConsoleMap.contains(pHost)) {
-            // get host profile display font and alter it, since that is how it's done in Settings.
-            QFont displayFont = pHost->mDisplayFont;
-            displayFont.setFamily(font);
-            pHost->mDisplayFont = displayFont;
+            if (!pHost->setDisplayFont(font)) {
+                lua_pushnil(L);
+                lua_pushstring(L, "specified font is invalid (average character width is 0)");
```

Comment:
This works for now, but show knowledge about the internal workings of `setDisplayFont()`. Can we return an error string from the function instead as well?

Reply:
There is no error string from the function though?

Reply:
That's right. This is why I'd move this message into `setDisplayFont` instead.

Reply:
I see what you mean.

Reply:
Added, have a look - but I'm not sure I like it. The other choice is to just return an `std::optional` which we now have access to, but that also doesn't seem ideal when you actually try using it...